### PR TITLE
Moved RPC related types to a dedicated package

### PR DESCRIPTION
**Pull request by vytautas-karpavicius**
**What changed?**
Moved RPC related stuff from `config` package to a dedicated `rpc` package, with minor renaming and splitting into files.

**Why?**
These types contain significant amount of RPC related logic, that IMO deserves their own place. Those are not very config related. 

Also, this serves as a first refactoring step for further changes planned. Specifically unifying how dispatchers are created and used. This includes `DispatcherProvider` and all related stuff such as `dnsDispatcherProvider`, `dnsUpdater`, etc. Currently they lie under `client` package. In the end, these types will end up here as well.

**How did you test it?**
Existing tests.

**Potential risks**
None, only mechanical move.

**Release notes**

**Documentation Changes**

### CODE REVIEW

**Review by meiliang86 — APPROVED** [review]

File: common/rpc/factory.go [hunk 1]
```diff
@@ -18,27 +18,26 @@
...
 )
 
-// RPCFactory is an implementation of service.RPCFactory interface
-type RPCFactory struct {
-	config      *RPC
+// Factory is an implementation of service.RPCFactory interface
+type Factory struct {
+	config      config.RPC
 	serviceName string
 	ch          *tchannel.ChannelTransport
```
Comment (mantas-sidlauskas) [inline common/rpc/factory.go:42]: nit and not directly related to this change: rename ch to tchannel?

File: common/rpc/factory.go [hunk 2]
```diff
@@ -49,19 +48,14 @@ type RPCFactory struct {
...
+// NewFactory builds a new rpc.Factory
 // conforming to the underlying configuration
-func (cfg *RPC) NewFactory(sName string, logger log.Logger, grpcPorts GRPCPorts) *RPCFactory {
-	return newRPCFactory(cfg, sName, logger, grpcPorts)
-}
-
-func newRPCFactory(cfg *RPC, sName string, logger log.Logger, grpcPorts GRPCPorts) *RPCFactory {
-	factory := &RPCFactory{config: cfg, serviceName: sName, logger: logger, grpcPorts: grpcPorts}
-	return factory
+func NewFactory(sName string, cfg config.RPC, logger log.Logger, grpcPorts GRPCPorts) *Factory {
```
Comment (mantas-sidlauskas) [inline common/rpc/factory.go:53]: nit and not directly related to this change: sName -> name or service

**Review by mantas-sidlauskas — APPROVED** [review]

File: common/rpc/factory.go (hunk 1 above)
Comment (vytautas-karpavicius) [reply common/rpc/factory.go]: Agree, fixed.

File: common/rpc/factory.go (hunk 2 above)
Comment (vytautas-karpavicius) [reply common/rpc/factory.go]: Agree, fixed.

**Pull request merged**